packages/simcore-sdk/src/simcore_sdk/node_ports_common/_filemanager.py

```diff
@@ -41,6 +41,13 @@ async def _get_location_id_from_location_name(
     raise exceptions.S3InvalidStore(store)
 
 
+def _get_https_link_if_storage_secure(url: AnyUrl) -> str:
+    if is_storage_secure() and not f"{url}".startswith("https"):
+        return url.replace("http", "https", 1)
+
+    return url
```

Comment:
this is very spaghetti if you ask me... but let's try that.

Reply:
no way to do that without string but with properl URL handling?

Reply:
why for instance all the calls to the storage client are not made https? how for example in get_storage_locations working? you do not use it?

Reply:
To shine some light on this. We have the following situation now.

```
WEBSERVER -> STORAGE (http requests)
DY-SIDECAR -> STORAGE (httpS requests)
```

The issue. Storage does not reply with the correct schema on some URLS. When it generates links that need to be called by other clients, it does not take into consideration if the request was coming from and HTTP or HTTPS endpoint (because before everything was http).

The only urls which require manipulation are the ones that storage provides back to calling clients.